react/src/components/PrometheusMetricGraph.tsx

```diff
@@ -0,0 +1,255 @@
+  endDate: string;
+  metricName: string;
+  fetchKey: string;
+}
+
+const PrometheusMetricGraph: React.FC<PrometheusMetricGraphProps> = ({
+  startDate,
+  endDate,
+  metricName,
+  fetchKey,
+}) => {
+  const { token } = theme.useToken();
+  const { styles } = useStyle();
+  const userInfo = useCurrentUserInfo();
+  const dayDiff = dayjs(Number(endDate) * 1000).diff(
+    dayjs(Number(startDate) * 1000),
+    'day',
+  );
+
+  const { user_utilization_metric } =
+    useLazyLoadQuery<PrometheusMetricGraphQuery>(
+      graphql`
+        query PrometheusMetricGraphQuery(
+          $user_id: UUID!
+          $props: UserUtilizationMetricQueryInput!
+        ) {
+          user_utilization_metric(user_id: $user_id, props: $props) {
+            user_id
+            metrics {
+              metric_name
+              value_type
+              values {
+                timestamp
+                value
+              }
+              max_value
+              avg_value
+            }
+          }
+        }
+      `,
+      {
+        user_id: userInfo[0]?.uuid ?? '',
+        props: {
+          metric_name: metricName,
+          start: startDate,
+          end: endDate,
+          step: dayDiff < 7 ? '5m' : dayDiff < 30 ? '1h' : '1d',
+        },
+      },
+      {
+        fetchPolicy: 'store-and-network',
+        fetchKey,
+      },
+    );
+
+  const metricData = getMetricData(
+    user_utilization_metric?.metrics ?? [],
+    startDate,
+    endDate,
+    dayDiff < 7 ? '5m' : dayDiff < 30 ? '1h' : '1d',
+  );
+
+  return (
+    <BAICard
+      title={_.startCase(metricName.replaceAll('_', ' '))}
+      type="inner"
+      styles={{
+        body: {
+          padding: `${token.marginMD}px ${token.marginMD}px ${token.marginXS}px ${token.marginMD}px`,
+        },
+      }}
+    >
+      {_.isEmpty(user_utilization_metric?.metrics) ? (
+        <Empty image={Empty.PRESENTED_IMAGE_SIMPLE} />
+      ) : (
+        <ResponsiveContainer width="100%" height={350}>
+          <LineChart data={metricData} className={styles.recharts}>
+            <CartesianGrid strokeDasharray="3 3" />
+            <XAxis dataKey="timestamp" minTickGap={token.marginMD} />
+            <YAxis domain={[0, 'dataMax']} />
+            <Tooltip
+              formatter={(value) => {
+                return `${value}${convertMetricUnit(undefined, metricName).numberUnit}`;
+              }}
+            />
+            <Legend />
+            <ReferenceLine
+              y={
+                convertMetricUnit(
+                  user_utilization_metric?.metrics?.[1]?.avg_value,
+                  user_utilization_metric?.metrics?.[1]?.metric_name,
+                ).number
+              }
+              label="Avg Used"
+              stroke={token.red}
+              strokeWidth={0.6}
+              strokeDasharray="6 6"
+            />
+            <Line
+              type="monotone"
+              dataKey="capacity"
+              stroke={token.colorLink}
```

Comment:
How about use colorSuccess or green? For now, capacity and used has very similar color.
![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/2HueYSdFvL8pOB5mgrUQ/b5efc1b5-45e5-4422-9341-8fe9a88969be.png)